toolkit/tools/scheduler/schedulerutils/depsolver.go

```diff
@@ -40,6 +48,22 @@ func CanSubGraph(pkgGraph *pkggraph.PkgGraph, node *pkggraph.PkgNode, useCachedI
 
 		// This node is not yet solvable
 		logger.Log.Warnf("Could not subgraph due to node: %v", pkgNode)
+
+		// If we are in trace mode, print the path from the root node to the unsolvable node
+		if logger.Log.IsLevelEnabled(logrus.TraceLevel) {
+			pt, ok := path.BellmanFordFrom(node, pkgGraph)
+			if !ok {
+				logger.Log.Warnf("Could not run BellmanFordFrom on %v", node)
+				return
+			} else {
+				path, _ := pt.To(pkgNode.ID())
+				logger.Log.Tracef("Path between %v and %v:", node, pkgNode)
+				for _, n := range path {
+					logger.Log.Warnf("  %v", n.(*pkggraph.PkgNode))
+				}
+			}
+		}
+
 		foundUnsolvableNode = true
```

Comment:
Since we return if `BellmanFordFrom` fails but that's only logging, I think we should set `foundUnsolvableNode` first. Also, I think we can drop `else` due to that `return`:
```suggestion
		foundUnsolvableNode = true

		// If we are in trace mode, print the path from the root node to the unsolvable node
		if logger.Log.IsLevelEnabled(logrus.TraceLevel) {
			pt, ok := path.BellmanFordFrom(node, pkgGraph)
			if !ok {
				logger.Log.Warnf("Could not run BellmanFordFrom on %v", node)
				return
			}

			path, _ := pt.To(pkgNode.ID())
			logger.Log.Tracef("Path between %v and %v:", node, pkgNode)
			for _, n := range path {
				logger.Log.Warnf("  %v", n.(*pkggraph.PkgNode))
			}
		}
```

Reply:
That was probably muscle memory tbh, I'm thinking we just ignore the failure and keep going.